lib/hanami/configuration.rb

```diff
@@ -61,6 +62,53 @@ def env
       settings.fetch(:env)
     end
 
+    def slices_dir=(dir)
+      settings[:slices_dir] = dir
+    end
+
+    def slices_dir
+      settings.fetch(:slices_dir) { "slices" }
+    end
+
+    def slices_namespace=(namespace)
+      settings[:slices_namespace] = namespace
+    end
+
+    def slices_namespace
+      settings.fetch(:slices_namespace) { Object }
+    end
+
+    # TODO: move into a "routing" namespace
+    def endpoint_resolver=(resolver)
+      settings[:endpoint_resolver] = resolver
+    end
+
+    def endpoint_resolver
+      settings.fetch(:endpoint_resolver) {
+        require_relative "web/endpoint_resolver"
+        Web::EndpointResolver
+      }
+    end
+
+    def action_key_namespace=(namespace)
+      settings[:action_key_namespace] = namespace
+    end
+
+    def action_key_namespace
+      settings.fetch(:action_key_namespace) { "actions" }
+    end
+
+    # TODO: need a better name for this, or to namespace it, e.g. web.logging.filter_params
+    DEFAULT_LOGGING_FILTER_PARAMS = %w[_csrf password password_confirmation].freeze
+
+    def logging_filter_params=(params)
+      settings[:logging_filter_params] = params
+    end
+
+    def logging_filter_params
+      settings[:logging_filter_params]
+    end
+
```

Comment:
n.b. I still need to settle on some better names (and placement/organisation of the methods) for these settings.